### ISSUE TITLE
Issue Reproducing Results

### ISSUE THREAD

**Issue by RamnathKumar181** [HungerPWAY/LoRA-Null#3]
Hi, firstly, I wanted to congratulate you on this great work!

I was actually curious about few discrepancies I have been observing in the results. I tried to run the baseline on pre-trained knowledge, and got slightly different numbers as shown below:

| Tasks  |Version|     Filter      |n-shot|  Metric   |   |Value |   |Stderr|
|--------|------:|-----------------|-----:|-----------|---|-----:|---|-----:|
|nq_open |      4|remove_whitespace|     0|exact_match|↑  |0.1845|±  |0.0065|
|triviaqa|      3|remove_whitespace|     0|exact_match|↑  |0.5054|±  |0.0037|
|webqs   |      2|none             |     0|exact_match|↑  |0.0792|±  |0.0060|

On a more serious note, I notice huge differences between the results I obtain upon running steps 1-4 and what was reported in the paper (Table 3(a)) with the LlaMA-2-7B model on MATH dataset:

| Tasks  |Version|     Filter      |n-shot|  Metric   |   |Value |   |Stderr|
|--------|------:|-----------------|-----:|-----------|---|-----:|---|-----:|
|nq_open |      4|remove_whitespace|     0|exact_match|↑  |0.0593|±  |0.0039|
|triviaqa|      3|remove_whitespace|     0|exact_match|↑  |0.4466|±  |0.0037|
|webqs   |      2|none             |     0|exact_match|↑  |0.0728|±  |0.0058|

On the math datasets, I have been observing a performance of 43.74 and 6.06% accuracy on the GSM8K and Math datasets respectively, which again is slightly different from your reported metrics. 
Please do let me know. Thanks again!

I used the same hyperparameters reported in the paper, and used in the codebase. Not sure why there is such a large deviation.

**Comment by HungerPWAY** [comment]
Thank you for your interest in our work. May I ask which model you are using? We are using the meta-llama/Llama-2-7b-hf model.

**Comment by RamnathKumar181** [comment]
Yes, even I am using the Llama-2-7b-hf model. Here is my script for getting baseline numbers:

```
CUDA_VISIBLE_DEVICES=0 accelerate launch -m lm_eval --model hf \
    --model_args pretrained="meta-llama/Llama-2-7b-hf" \
    --output_path baseline/result.json \
    --tasks  triviaqa,webqs,nq_open\
    --batch_size 8 \
    --max_batch_size 16 \
    --device cuda
```

I used the steps1-4 as is with little to no modifications as well to remain faithful to your implementation.

**Comment by HungerPWAY** [comment]
This is the result we just tested for base model. We used version 0.4.9.1 of lm_eval.
| Tasks  |Version|     Filter      |n-shot|  Metric   |   |Value |   |Stderr|
|--------|------:|-----------------|-----:|-----------|---|-----:|---|-----:|
|nq_open |      4|remove_whitespace|     0|exact_match|↑  |0.1886|±  |0.0065|
|triviaqa|      3|remove_whitespace|     0|exact_match|↑  |0.5252|±  |0.0037|
|webqs   |      2|none             |     0|exact_match|↑  |0.0586|±  |0.0052|

**Comment by RamnathKumar181** [comment]
Hey, sorry for the delayed response. Thank you very much! With the version change to 0.4.9.1 of lm_eval, I am able to get the exact metrics you reported for the baseline. I am also getting the following numbers on the LoRA-Null approach:

| Tasks  |Version|     Filter      |n-shot|  Metric   |   |Value |   |Stderr|
|--------|------:|-----------------|-----:|-----------|---|-----:|---|-----:|
|nq_open |      4|remove_whitespace|     0|exact_match|↑  |0.0778|±  |0.0045|
|triviaqa|      3|remove_whitespace|     0|exact_match|↑  |0.5101|±  |0.0037|
|webqs   |      2|none             |     0|exact_match|↑  |0.0694|±  |0.0056|

I was curious about the slight discrepancy on the MATH benchmarks though - 43.74% and 6.06% compared to the reported 44.43% and  8.80% respectively. Thanks again!

**Comment by HungerPWAY** [comment]
Due to differences in random factors such as the machine and random seeds, the results may vary slightly. This is a normal phenomenon. Please refer to Table 8 in our paper https://arxiv.org/pdf/2503.02659v2 , which shows the results using different random seeds.
